curie/experiment.py

```diff
@@ -316,6 +317,29 @@ def validate_input(question_file: Optional[str],
         raise ValueError(f"Question file {question_file} does not exist.") 
     return 
 
+def split_code_instruction_from_question(codebase_dir: str, question: Optional[str]) -> Optional[str]:
+    if question is None or "<CODE INSTRUCTION>" not in question:
+        return None
+
+    try:
+        parts = question.split("<CODE INSTRUCTION>", 1)
```

Comment:
need to update the code to fit the new pattern. please run the code e2e to verify your implementation